### PR TITLE
Split up sequence assignments generated by decaf

### PR DESCRIPTION
**Pull request by GoodForOneFare**
`decaf` takes stuff like `{@foo, @bar} = opts` and turns it into a sequence like `foo = opts.food, bar = opts.bar, opts`.  This PR:
* Converts each assignment to its own expression
* Removes the trailing `opts`

Note, I haven't handled some edge cases because they have no value to Shopify's current codebase:
* Transforming `return foo = opts.foo, opts` to `foo = opts.foo; return opts`
* Checking that all assigned fields are identifiers with a name matching their source property

Please review @lemonmade, @TylerHorth

### CODE REVIEW

File: packages/shopify-codemod/docs/split-assignment-sequences.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+```sh
+jscodeshift -t shopify-codemod/transforms/split-assignment-sequences <file>
+```
+
+#### Example
+
+```js
+foo = options.foo, bar = options.bar, options;
+
+this.foo = opts.foo, bar = opts.bar, options;
```
Comment (lemonmade) [inline packages/shopify-codemod/docs/split-assignment-sequences.md]: Trailing `opts`?

File: packages/shopify-codemod/transforms/split-assignment-sequences.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+      const lastExpression = expressions[expressions.length - 1];
+      const otherExpressions = expressions.slice(0, -1);
+
+      return j.Identifier.check(lastExpression) &&
+        otherExpressions.every(j.AssignmentExpression.check) &&
+        otherExpressions.every((expression) =>
+          expression.right.object.name === lastExpression.name,
+        );
+    })
+    .replaceWith((path) => {
```
Comment (lemonmade) [inline packages/shopify-codemod/transforms/split-assignment-sequences.js]: `forEach` and call `replace(null)` at the end? It feels awkward to prepend in `replaceWith`.

**Comment by TylerHorth** [comment]
Is it possible to translate this to object destructuring?

----
```js
foo = opts.foo, bar = opts.bar, opts
```
=>
```js
({foo, bar} = opts);
```
----
```js
this.foo = opts.foo, this.bar = opts.bar, opts
```
=>
```js
({foo: this.foo, bar: this.bar} = opts);
```

**Comment by GoodForOneFare** [comment]
@lemonmade examples of Coffee->decaf->JS that generates these:

`card_consumption`:
```coffee
class Shopify.CardConsumption
  constructor: (options) ->
    { @requestId } = options
```

becomes:

```js
    this.requestId = options.requestId, options;
```

`notifications.coffee` contains `{@iconPath, @shopName} = opts`

`inventory_options` contains 
```coffee
   {
      @onAdjustmentUpdated,
      @onQuantityUpdated,
      @onAdjustmentIncremented,
      @onAdjustmentDecremented
    } = options
```

**Comment by GoodForOneFare** [comment]
@TylerHorth that's possible, but I find the es5 output to be way more readable ¯\_(ツ)_/¯  e.g.,

```js
this.foo = opts.foo;
this.bar = opts.bar;
```

... has the same character count as the deconstructed version, but I guarantee you that anyone in #fed will understand the es5 version in half the time.

**Comment by GoodForOneFare** [comment]
@lemonmade please take another look.  Example Coffee provided, docs fixed, replace/insert/null changed to forEach/insert/replace.

**Pull request merged**